lib/plugins/aws/deploy/lib/createStack.js

```diff
@@ -28,39 +30,68 @@ module.exports = {
       stackTags = Object.assign(stackTags, this.serverless.service.provider.stackTags);
     }
 
-    const params = {
+    const createChangeSetParams = {
       StackName: stackName,
-      OnFailure: 'DELETE',
+      ChangeSetName: changeSetName,
+      ChangeSetType: 'CREATE',
       Capabilities: ['CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM'],
       Parameters: [],
       TemplateBody: JSON.stringify(this.serverless.service.provider.coreCloudFormationTemplate),
       Tags: Object.keys(stackTags).map((key) => ({ Key: key, Value: stackTags[key] })),
     };
 
+    const executeChangeSetParams = {
+      StackName: stackName,
+      ChangeSetName: changeSetName,
```

Comment:
In so far version we've setup `OnFailure: 'DELETE'`, so in case of stack creation failure, stack is removed.

What will happen with changesets?

Reply:
It's not possible to set something like that with change sets - in such situations, there will be stack left that has 0 resources deployed - if will rollback to an empty stack. We can potentially remove it, but thanks to changes with ensuring that deployment bucket exists and improvements to `remove`, user can safely continue with subsequent `deploy` or a `remove` to cleanup the stack fully

Reply:
Indeed, I believe we discuss it in the past, and then agreed to first fix the logic in v2 that ensures that deployment bucket exists, if we find empty stack